### PR TITLE
Add init script

### PR DESCRIPTION
**Pull request by ynachiket**
Why: 

While using supervisord the noit and stratcon processes were being started pretty much simultaneously which made a curl http://noit/handoff/journals -k -v from a remote container not receive any journals.

Solution: 
A bash script with a small sleep between the 2 processes. 

Workaround:
The issue with using a shell script with Docker is that as soon as the script is done executing the container exits. Even running with a -d flag does not help as the entry point itself exits.
running with -t -i  /bin/bash does not start the processes in the container.

So added a
CMD /bin/bash noitstratconinit.bash && tail -f /var/log/noitd.feed/00000000 and ran container with -d flag.

To test this out 

docker run -i -t  -P --name postgres -d nachiket/postgres
docker run --name apache_borrowed -p 80:80 -p 443:443 -d eboraas/apache
docker run -d -p 32322:32322 -p 43191:43191 -p 8888:8888 -p 80:80 -ti --link postgres:postgres --link apache_borrowed:apache --name noit nachiket/noit

If you are on mac
curl http://192.168.59.103/handoff/journals (make sure you boot2docker ip is actually 192.168.59.103 )

If you are on any Linux based OS
curl http://localhost/handoff/journals

Note: You will start seeing the journals in about 3 to 5 sec after the hello message.

### CODE REVIEW

**Comment by robert-chiniquy** [comment]
> To test this out

I don't understand. What does this test? Why is it not an actual test?

**Comment by robert-chiniquy** [comment]
Why is `conf/stratcon.conf` in `conf/`? It is a dependency right, maybe this `conf/` dir should be in `deps/`?

File: noitstratconinit.bash [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+/usr/local/sbin/noitd -c noitconf/noit.conf
```
Comment (robert-chiniquy) [inline noitstratconinit.bash]: I guess noit daemonizes itself?

**Comment by robert-chiniquy** [comment]
So with `noitstratconinit.bash`, what's the plan for process management? After these processes are started, how do we stop them?

**Comment by robert-chiniquy** [comment]
Should this `Dockerfile` be at the top level of this repo or should it be in some noit-specific dir under `deps/`? How many containers will this project ultimately have?

File: noitstratconinit.bash (hunk 1 above)
Comment (ynachiket) [inline noitstratconinit.bash]: yes.

**Comment by ynachiket** [comment]
@robert-chiniquy 

In my mind this is how things will work.

npm init : 
run noit/stratcon in a container (this container will be linked to apache, postgres containers)
run similar setup for agent in another container

npm test
use tape to create tests which will create checks against the noit using noit_client and agent-poller-mode
using agent equivalent of noit_client
use the stratcon client and equivalent for agent to test parity

as of now i was focusing on building the blocks. will add the test structure next.

**Comment by robert-chiniquy** [comment]
`npm init` <—— https://www.npmjs.org/doc/cli/npm-init.html

**Comment by robert-chiniquy** [comment]
`Dockerfile → deps/Dockerfile` <—— cool

File: deps/Dockerfile [hunk 2]
```diff
@@ -4,20 +4,20 @@ RUN apt-get update && \
...
  libapr1-dev libaprutil1-dev xsltproc libncurses5-dev python \
- libssh2-1-dev libsnmp-dev openjdk-6-jdk libprotobuf-c0-dev uuid-runtime telnet vim git lsof supervisor
+ libssh2-1-dev libsnmp-dev openjdk-6-jdk libprotobuf-c0-dev uuid-runtime telnet vim git lsof
 
 
-ADD deps/noit /opt/noit
+ADD noit /opt/noit
 ADD conf /noitconf
-ADD conf/supervisord.conf /etc/supervisor/conf.d/supervisord.conf
 RUN mkdir "/var/log/noitd.feed(stratcon)"
```
Comment (robert-chiniquy) [inline deps/Dockerfile:12]: the directory has `()` in its name?

**Comment by robert-chiniquy** [comment]
Cool. Looks a lot better. What does this need for a +1?

File: deps/Dockerfile (hunk 2 above)
Comment (ynachiket) [inline deps/Dockerfile:12]: Yea it is weird but i was just following what it says 
https://github.com/circonus-labs/reconnoiter/blob/master/src/noit.conf.in#L21

**Comment by ynachiket** [comment]
@robert-chiniquy 
i think it is good to go.

Next up: use depends-on and figure out running docker with it

**Comment by lekhajee** [comment]
+1

**Pull request merged**